portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -58,29 +58,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    BufferedReader body = request.getReader();
-    String content = body.readLine();
-    int endOfNameIndex = content.indexOf("*");
-    if (endOfNameIndex != 0 && endOfNameIndex != (content.length() -1 )) {
-      // Get name and text
-      String name = content.substring(0, endOfNameIndex++);
-      String text = content.substring(endOfNameIndex);
+    // Get name and text
+    String name = request.getParameter("name-input");
+    String text = request.getParameter("text-input");
 
-      long timestamp = System.currentTimeMillis();
+    long timestamp = System.currentTimeMillis();
 
-      Entity messageEntity = new Entity("Messages");
-      messageEntity.setProperty("name", name);
-      messageEntity.setProperty("text", text);
-      messageEntity.setProperty("timestamp", timestamp);
+    Entity messageEntity = new Entity("Messages");
```

Comment:
This is common across two classes. A fairly routine problem people face when writing code is getting their string-constants to match (and often they just mess it up by using a capital or non-capital, or underscores rather than dashes etc.)

There's a fun joke in CS: "as you get to be a better engineer, a higher and higher percentage of your bugs will be typos".

To prevent that, you could create a Constants.java file that contains the name of the message type:

public final class Constants {
  public static final String MESSAGE_ENTITY_TYPE = "Messages";
}

then use that across the board in place of hardcoded values.

Reply:
done